Source/Scene/ModelExperimental/LightingPipelineStage.js

```diff
@@ -0,0 +1,36 @@
+import ShaderDestination from "../../Renderer/ShaderDestination.js";
+import LightingStageFS from "../../Shaders/ModelExperimental/LightingStageFS.js";
+
+/**
+ * The lighting pipeline stage is responsible for taking a material and rendering
+ * it with a lighting model such as physically based rendering (PBR) or unlit
+ * shading
+ * @namespace
+ * @private
+ */
+var LightingPipelineStage = {};
+
+/**
+ * Process a primitive. This modifies the following parts of the render
+ * resources:
+ * <ul>
+ *   <li>modifies the shader to include the lighting stage</li>
+ * </ul>
+ * @param {PrimitiveRenderResources} renderResources The render resources for the primitive
+ * @param {ModelComponents.Primitive} primitive The primitive to be rendered. Not used, but present for consistency with other pipeline stages.
```

Comment:
For the `InstancingPipelineStage`, we need to pass in the `frameState` too. Do we always mention in the documentation when a parameter is unused?

Reply:
Hm, this is a good question. Though in this case, Instancing is a different type of pipeline stage because it takes a NodeRenderResources, right?

:open_mouth:  oh this should be `RenderResources.PrimitiveRenderResources` I'll fix that much for now.